### PR TITLE
doc: move pkg/PORTING.md and pkg/USING.md to doxygen

### PR DESCRIPTION
**Pull request by authmillenon**
Moves the documentation for packages to doxygen.

### CODE REVIEW

**Comment by OlegHahm** [comment]
Didn't you say that doxygen can parse markdown as well?

**Comment by authmillenon** [comment]
It can parse it, yes. But it generates automatically pages for every new markdown file (named the name of the markdown file). We don't want this here, but just a module documentation.

In detail: when adding a `doc.md` instead of a `doc.txt` (and adding `doc.md` to the FILE_PATTERN config var of doxygen), the page tree of doxygen looks like:

```
▼ RIOT Documentation
   doc                       # containing an empty page
   RIOT in a nutshell
   Contribute to RIOT
   The quickest start
   Structure
   Todo List
 ▶ Deprecated List
Modules
 ▶ Boards
 ▶ CPU
 ▶ Drivers
 ▶ Kernel
 ▶ Networking
   Packages
 ▶ System
▶ Data Structures
▶ Files
```

I don't want this empty `doc` page ;-)

**Comment by OlegHahm** [comment]
I see.

**Comment by gebart** [comment]
A somewhat related configuration option I came across by accident: 
See http://stackoverflow.com/questions/13368350/use-the-readme-md-file-as-main-page-in-doxygen and 
https://github.com/authmillenon/RIOT/blob/d50de568d1c00162900d9ba478f67790b1b10091/doc/doxygen/riot.doxyfile#L924
http://superuser.com/questions/685817/doxygen-specify-a-markdown-file-as-the-main-page

**Comment by authmillenon** [comment]
I saw this, too. But since PRs like https://github.com/RIOT-OS/RIOT/pull/4285 will introduce further pages, I thought it to be less consistent. Additionally, the doxygen markdown still requires labels to be of any use to doxygen (Sub-sections in page tree, TOC, and ability to reference sections or pages from within doxygen). This isn't parsed however by GitHub properly (as well as the `@ref` references to other pages). Have a look at [the main page on GitHub](https://github.com/RIOT-OS/RIOT/blob/master/doc/doxygen/src/mainpage.md) vs the [intended result](http://doc.riot-os.org) to see what I mean.

**Comment by authmillenon** [comment]
Rebased to current master

File: doc/doxygen/riot.doxyfile [hunk 1]
```diff
@@ -758,6 +758,7 @@ INPUT                  = ../../doc.txt \
                          ../../cpu \
                          ../../boards \
                          ../../drivers \
+                         ../../pkg \
```
Comment (OlegHahm) [inline doc/doxygen/riot.doxyfile]: duplicate line

Comment (authmillenon) [inline doc/doxygen/riot.doxyfile]: Fixed and squashed immediately

File: pkg/doc.txt [hunk 2]
```diff
@@ -1,15 +1,44 @@
...
+ *
+ * Using packages
+ * ==============
+ * To add a package to the list of compiled modules you have to add it to the
+ * `USEPKG` macro in your application's Makefile. If the package provides
+ * header files you might need to update the `INCLUDE` macro, too:
+ *
+ * ~~~~~~~~ {.mk}
+ * USEPKG += <pkg_name>
+ * INCLUDE += $(RIOTBASE)/pkg/<pkg_name>/...
```
Comment (Lotterleben) [inline pkg/doc.txt]: maybe `INCLUDE += $(RIOTBASE)/pkg/<pkg_name>/<path to include directory>` could be even more explicit, or is that overkill?

Comment (authmillenon) [inline pkg/doc.txt]: I think that's overkill. Also the `pkg` directory might be the include directory.

**Comment by OlegHahm** [comment]
ACK and go!

**Pull request merged**